lib/posix/c_lib_ext/fnmatch.c

```diff
@@ -48,6 +48,159 @@
 
 #define EOS '\0'
 
+#define MATCH_CLASS6(p, a, b, c, d, e, f, g)                                                       \
+	((p)[0] == (a) && (p)[1] == (b) && (p)[2] == (c) && (p)[3] == (d) && (p)[4] == (e) &&      \
+	 (p)[5] == (f) && (p)[6] == (g))
+
+#define MATCH_CLASS7(p, a, b, c, d, e, f, g, h)                                                    \
+	((p)[0] == (a) && (p)[1] == (b) && (p)[2] == (c) && (p)[3] == (d) && (p)[4] == (e) &&      \
+	 (p)[5] == (f) && (p)[6] == (g) && (p)[7] == (h))
+
+enum fnm_char_class {
+	FNM_CC_ALNUM,
+	FNM_CC_ALPHA,
+	FNM_CC_CNTRL,
+	FNM_CC_DIGIT,
+	FNM_CC_GRAPH,
+	FNM_CC_LOWER,
+	FNM_CC_PRINT,
+	FNM_CC_PUNCT,
+	FNM_CC_SPACE,
+	FNM_CC_UPPER,
+	FNM_CC_XDIGIT,
+	FNM_CC_INVALID,
+};
+
+static bool fnm_cc_is_valid(const char *pattern, size_t psize, enum fnm_char_class *cc)
+{
+	if (psize < 4 || *pattern != ':') {
+		return false;
+	}
+
+	pattern++; /* skip ':' */
+	psize--;
+
+	/* Each class name ends with ":]" */
+	switch (pattern[0]) {
+	case 'a':
+		if (MATCH_CLASS6(pattern, 'a', 'l', 'n', 'u', 'm', ':', ']')) {
+			*cc = FNM_CC_ALNUM;
+			return true;
+		}
+		if (MATCH_CLASS6(pattern, 'a', 'l', 'p', 'h', 'a', ':', ']')) {
+			*cc = FNM_CC_ALPHA;
+			return true;
+		}
+		break;
+
+	case 'c':
```

Comment:
Again for `[:blank:]` handling.
```suggestion
		case 'b':
			if (MATCH_CLASS6(pattern, 'b', 'l', 'a', 'n', 'k', ':', ']')) {
				*cc = FNM_CC_BLANK;
				return true;
			}
			break;

		case 'c':
```